src/studio/src/designer/backend/Repository/DeploymentRepository.cs

```diff
@@ -1,23 +1,167 @@
+using Altinn.Studio.Designer.ViewModels.Request;
+using Altinn.Studio.Designer.ViewModels.Request.Enums;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Npgsql;
+using NpgsqlTypes;
 
 namespace Altinn.Studio.Designer.Repository
 {
     /// <summary>
-    /// DeploymentRepository
+    /// Handles deployment repository. 
     /// </summary>
-    public class DeploymentRepository : DocumentRepository
+    [ExcludeFromCodeCoverage]
+    public class DeploymentRepository : IDeploymentRepository
     {
+        private readonly string insertDeploymentSql = "call designer.insert_deployment(@id, @tagName, @org, @app, @buildId, @buildResult, @created, @entity)";
+        private readonly string getDeploymentsSql = "select designer.get_deployments(@_org, @_app, @_limit, @_order_asc_desc)";
+        private readonly string getDeploymentSql = "select designer.get_deployment(@_org, @_buildId)";
+        private readonly string updateDeploymentBuildSql = "call designer.update_deployment_build(@_id, @_buildResult, @_entity)";
+        private readonly string _connectionString;
+        private readonly ILogger _logger;
+
         /// <summary>
-        /// DeploymentRepository
+        /// Initializes a new instance of the <see cref="DeploymentRepository"/> class.
         /// </summary>
-        public DeploymentRepository(
-            IOptions<AzureCosmosDbSettings> options,
-            IDocumentClient documentClient)
-            : base(options.Value.DeploymentCollection, options, documentClient)
+        public DeploymentRepository(IOptions<PostgreSQLSettings> postgresSettings, ILogger<DeploymentRepository> logger)
+        {
+            _connectionString = string.Format(
+                postgresSettings.Value.ConnectionString,
+                postgresSettings.Value.DesignerDbPwd);
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task<DeploymentEntity> Create(DeploymentEntity deploymentEntity)
+        {
+            try
+            {
+                using NpgsqlConnection conn = new NpgsqlConnection(_connectionString);
+                await conn.OpenAsync();
+
+                NpgsqlCommand pgcom = new NpgsqlCommand(insertDeploymentSql, conn);
+                pgcom.Parameters.AddWithValue("id", deploymentEntity.Id);
+                pgcom.Parameters.AddWithValue("tagName", deploymentEntity.TagName);
+                pgcom.Parameters.AddWithValue("org", deploymentEntity.Org);
+                pgcom.Parameters.AddWithValue("app", deploymentEntity.App);
+                pgcom.Parameters.AddWithValue("buildId", deploymentEntity.Build.Id);
+                pgcom.Parameters.AddWithValue("buildResult", deploymentEntity.Build.Result.ToEnumMemberAttributeValue());
+                pgcom.Parameters.AddWithValue("created", deploymentEntity.Created);
+                pgcom.Parameters.AddWithValue("entity", JsonString(deploymentEntity));
```

Comment:
Is this identical for the releases and deployments? could we find a way to limit the duplications?

Reply:
No, they are not identical.